atmat/lattice/atfitchrom.m

```diff
@@ -13,19 +15,16 @@
 %   Numeric array: list of selected elements in RING
 %   Cell array: All elements selected by each cell
 
-if isscalar(varargin{1}) && isnumeric(varargin{1})
-    dpp=varargin{1};
-    [newchrom,famname1,famname2]=deal(varargin{2:end});
-else
-    dpp=0;
-    [newchrom,famname1,famname2]=deal(varargin{:});
-end
-deltaP = 1e-8;
+check_radiation(ring,false);
+[deltaP,varargin]=getoption(varargin,'DPStep');
+[dpp,varargin]=getargs(varargin,0.0,'check',@(arg) isscalar(arg) && isnumeric(arg));
+[newchrom,famname1,famname2]=deal(varargin{:});
+
 idx1=varelem(ring,famname1);
 idx2=varelem(ring,famname2);
 kl1=atgetfieldvalues(ring(idx1),'PolynomB',{3});
 kl2=atgetfieldvalues(ring(idx2),'PolynomB',{3});
-if true
+%if true
     deltaS = 1e-5; % step size in Sextupole strngth
```

Comment:
Also this deltaS could be treated as the DPStep? set as optional input and defaulted to the present value

Reply:
Done